custom_components/xiaomi_miio/coordinator.py

```diff
@@ -16,10 +16,15 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+ALLOWED_RETRY_COUNT = 3
+
 
 class XiaomiDataUpdateCoordinator(DataUpdateCoordinator):
     """Update coordinator for xiaomi_miio."""
 
+    retry_count = -1
```

Comment:
Maybe inverse the logic and use `retries_available`? Or even better, just in case we adjust the update frequency, something like `last_known_state_ttl` and rename `saved_state` as `last_known_state`?

Reply:
`retry_count` member defined here is set to `ALLOWED_RETRY_COUNT` on fetch success, and then counted down for each failed retry.
0 means that it ran out of retries.
-1 means that it never succeeded to fetch data, set initially.

I don't think that reverting the logic to count up is necessary, especially when in xiaomi-miio is using the same counting down logic.

Reply:
Yeah, I agree that it doesn't make a big difference whichever way it's done... Which made me wonder, perhaps it would make sense to use the cached status for a timespan of X (30s? 60s? how long does it take for the device to respond again?) just when the device is not responsive?

The `UPDATE_INTERVAL` might be something we may want to adjust in the future, so if we'd just store `last_known_state` and `last_known_state_timestamp` (or similar), we can easily change the behavior in the future if needed. This would keep the logic clean & leave it up to homeassistant to perform "retries" which is probably a good idea anyway as the lib is not using asyncio.

Reply:
I noticed that my Xiaomi device took around 12 secods to reconnect to wifi network. Initially I wanted to have 2 retries but for some reasons it still gave some unavailable states in home assistant.